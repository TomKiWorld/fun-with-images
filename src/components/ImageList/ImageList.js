import React from 'react';
import ImageCard from '../ImageCard/ImageCard';
import Wiggle from '..//Wiggle/Wiggle';

const ImageList = ({ images , onResubmit}) => {
  return (
    <div>
      {
        images.map(image => {
          return (
            <Wiggle
              key={image.id}
              elementId= {`ImageCard-${image.id}`}>
                <ImageCard
                  key={image.id}
                  id={image.id}
                  url={image.url}
                  onResubmit={onResubmit}
                />
            </Wiggle>
          );
        })
      }
    </div>
  );
}

export default ImageList;
