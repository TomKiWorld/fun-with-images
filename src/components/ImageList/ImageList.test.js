import { shallow } from 'enzyme';
import React from 'react';
import ImageList from './ImageList';

describe('ImageList Component', () => {
  const mockImages = [
    {
      id: 10,
      url: 'https://i.pinimg.com/originals/29/80/94/298094c2d1dbf6267d5e61d741b902c2.jpg'
    }
  ];

  it('Expect to render ImageList Component', () => {
    expect(shallow(<ImageList images={mockImages}/>)).toMatchSnapshot();
  });
});
