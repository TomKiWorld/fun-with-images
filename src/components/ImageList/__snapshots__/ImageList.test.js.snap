// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList Component Expect to render ImageList Component 1`] = `ShallowWrapper {}`;
